types/types.go

```diff
@@ -5,6 +5,8 @@ import "time"
 // ImageBuilder is the interface for building Docker images.
 type ImageBuilder interface {
 	Build(opts ImageBuilderOpts) error
+	Pull(tag string) error
+	Retag(from, to string) error
```

Comment:
In my opinion these methods should not be part of the `ImageBuilder` interface.
The interface was designed to abstract the image build backend. The backend could be for instance `docker`, `kaniko`, `buildkit` etc...

It does not make any sense to do a `Pull` or a `Retag` with kaniko

Also, best practices recommend to keep the interfaces as small as possible, like explained in this article: https://www.integralist.co.uk/posts/go-interfaces/#keep-interfaces-small

I suggest we create a new interface `ImageTagger` which would hold the method `Retag` (or simply `Tag`), with multiple implementations:
- one to do the tag with `docker`
- one to tag directly on the remote registry. We currently have `registry.Registry` which pretty much implements the interface already.
- eventually an other one in the `mock` package if we need for tests

The `Pull` method doesn't need to be exposed as it is an implementation detail of the "local" `Tag` with `docker`.

Then, we would simply inject the right instance when initializing the `DAG`, depending on the `--local-only` flag.

Reply:
Done